subprojects/maven/src/main/java/org/gradle/api/publish/maven/tasks/PublishToMavenRepository.java

```diff
@@ -96,6 +112,15 @@ private PublishSpec computeSpec() {
             throw new InvalidUserDataException("The 'repository' property is required");
         }
 
+        Property<Credentials> credentialsProperty = repository.getConfiguredCredentials();
+        Credentials value = credentialsProperty.getOrNull();
+
+        boolean safeCredentials = value == null || areCredentialsSafe(repository.getName(), value);
+
+        if (!safeCredentials && getServices().get(BuildModelParameters.class).isConfigurationCache()) {
```

Comment:
Is checking for --configuration-cache necessary? Or is the listener only there if CC is enabled, and then we can blindly fire the event?

Reply:
We can blindly fire the event.